### PR TITLE
Fix for possible request starvation for multiple partitions consumer

### PR DESCRIPTION
**Pull request by ozangunalp**

### CODE REVIEW

File: smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/perf/PauseResumeBatchPerfTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        long start = System.currentTimeMillis();
+        await()
+                .atMost(Duration.ofSeconds(TIMEOUT_IN_SECONDS))
+                .until(() -> application.getCount() == COUNT);
+
+        long end = System.currentTimeMillis();
+
+        assertThat(application.get()).containsExactlyInAnyOrderElementsOf(expected);
+
+        System.out.println("Post acknowledgement (auto-commit, requests, multiple partitions) - Estimate: " + (end - start) + " ms");
```
Comment (cescoffier) [inline smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/perf/PauseResumeBatchPerfTest.java:87]: Same

File: smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/perf/PauseResumeBatchPerfTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import org.junit.jupiter.api.Test;
+
+import io.smallrye.reactive.messaging.kafka.TestTags;
+import io.smallrye.reactive.messaging.kafka.base.KafkaCompanionTestBase;
+import io.smallrye.reactive.messaging.kafka.base.KafkaMapBasedConfig;
+import io.smallrye.reactive.messaging.kafka.companion.ProducerTask;
+
+@Tag(TestTags.PERFORMANCE)
+@Tag(TestTags.SLOW)
+public class PauseResumeBatchPerfTest extends KafkaCompanionTestBase {
```
Comment (cescoffier) [inline smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/perf/PauseResumeBatchPerfTest.java:34]: Is it really a perf test?

**Review by cescoffier — CHANGES_REQUESTED** [review]
Just the traces to remove.
The code looks good.

File: smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/perf/PauseResumeBatchPerfTest.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        expected = new ArrayList<>();
+        companion.topics().createAndWait(topic, partitions);
+        ProducerTask recordMetadata = companion.produceStrings().usingGenerator(i -> {
+            expected.add(Long.toString(i));
+            return new ProducerRecord<>(topic, "k-" + i, Long.toString(i));
+        }, COUNT).awaitCompletion(Duration.ofMinutes(2));
+        Map<TopicPartition, Long> latestOffsets = recordMetadata.byTopicPartition().entrySet().stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().stream()
+                        .mapToLong(RecordMetadata::offset).max().orElse(-1L)));
+        System.out.println(latestOffsets);
```
[the remark this answers is not in the document]
Comment (ozangunalp) [reply smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/perf/PauseResumeBatchPerfTest.java:55]: Done

File: smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/perf/PauseResumeBatchPerfTest.java (hunk 2 above)
Comment (ozangunalp) [reply smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/perf/PauseResumeBatchPerfTest.java]: I moved it to client tests along side `PauseResumeTest`

**Review by cescoffier — APPROVED** [review]

**Pull request merged**